cli/src/pcluster/config/cluster_config.py

```diff
@@ -1483,10 +1504,120 @@ def _register_validators(self):
         )
 
 
-class ByosQueue(SlurmQueue):
+class ByosComputeResource(SlurmComputeResource):
+    """Represent the Byos Compute Resource."""
+
+    def __init__(
+        self,
+        instance_types: List[str],
+        custom_settings: Dict = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.instance_types = instance_types
+        self.custom_settings = custom_settings
+
+
+class ByosQueue(_CommonQueue):
     """Represent the Byos queue."""
 
-    pass
+    def __init__(
+        self,
+        compute_resources: List[ByosComputeResource],
+        networking: ByosQueueNetworking,
+        custom_settings: Dict = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.compute_resources = compute_resources
+        self.networking = networking
+        self.custom_settings = custom_settings
+
+    def _register_validators(self):
+        super()._register_validators()
+        self._register_validator(DuplicateInstanceTypeValidator, instance_type_list=self.instance_type_list)
+        self._register_validator(
+            DuplicateNameValidator,
+            name_list=[compute_resource.name for compute_resource in self.compute_resources],
+            resource_name="Compute resource",
+        )
+        for compute_resource in self.compute_resources:
+            self._register_validator(
+                CapacityTypeValidator, capacity_type=self.capacity_type, instance_type=compute_resource.instance_type
+            )
+            self._register_validator(
+                EfaSecurityGroupValidator,
+                efa_enabled=compute_resource.efa.enabled,
+                security_groups=self.networking.security_groups,
+                additional_security_groups=self.networking.additional_security_groups,
+            )
+            self._register_validator(
+                EfaPlacementGroupValidator,
+                efa_enabled=compute_resource.efa.enabled,
+                placement_group_enabled=self.networking.placement_group and self.networking.placement_group.enabled,
+                placement_group_config_implicit=self.networking.placement_group is None
+                or self.networking.placement_group.is_implied("enabled"),
+            )
+
+    @property
+    def instance_type_list(self):
```

Comment:
can this be moved to _CommonQueue?

Reply:
No, ComputeResource is not defined in _CommonQueue